front/lib/resources/vault_resource.ts

```diff
@@ -30,14 +32,32 @@ export interface VaultResource extends ReadonlyAttributesType<VaultModel> {}
 export class VaultResource extends BaseResource<VaultModel> {
   static model: ModelStatic<VaultModel> = VaultModel;
 
-  constructor(model: ModelStatic<VaultModel>, blob: Attributes<VaultModel>) {
+  // readonly groups: GroupResource[];
+
+  constructor(
+    model: ModelStatic<VaultModel>,
+    blob: Attributes<VaultModel>,
+    public readonly groups: GroupResource[]
+  ) {
     super(VaultModel, blob);
+    this.groups = groups;
   }
 
-  static async makeNew(blob: CreationAttributes<VaultModel>) {
+  static async makeNew(
+    blob: CreationAttributes<VaultModel>,
+    groups: GroupResource[]
```

Comment:
Can we only accept one group for now? Even though from a data model standpoint we support many groups, the product is only aware of one group today.